src/promptflow/promptflow/_utils/utils.py

```diff
@@ -117,10 +117,22 @@ def count_and_log_progress(
         yield item
 
 
-def log_progress(logger: logging.Logger, count: int, total_count: int, formatter="{count} / {total_count} finished."):
+def log_progress(
+    start_time: datetime,
+    logger: logging.Logger,
+    count: int,
+    total_count: int,
+    formatter="{count} / {total_count} finished.",
+):
     log_interval = max(int(total_count / 10), 1)
     if count % log_interval == 0 or count == total_count:
+        average_execution_time = round((datetime.now().timestamp() - start_time.timestamp()) / count, 2)
```

Comment:
one silly question: is it possible for `count` be zero? as I always encounter this issue when calculate average metrics...

Reply:
From the internal logic of the executor, this cannot be 0. But it is better to add a judgment just in case.